homeassistant/components/shelly/entity.py

```diff
@@ -19,6 +27,41 @@ async def async_setup_entry_attribute_entities(
     wrapper: ShellyDeviceWrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][
         config_entry.entry_id
     ][COAP]
+
+    if wrapper.restored_device:
```

Comment:
`async_setup_entry_attribute_entities` should be split in two. This if-statement already makes it have two different implementations anyway.